File: test/test_vit_custom.py
import os
import unittest
import torch
import numpy as np
from transformers import AutoImageProcessor, AutoModel

# Hardware detection
HAS_CUDA = torch.cuda.is_available()
HAS_WEBGPU = "WEBGPU_AVAILABLE" in os.environ

class TestVit(unittest.TestCase):
    def setUp(self):
        self.model_name = "vit"
        self.dummy_image = np.random.rand(3, 224, 224)
    
    def test_cpu(self):
        processor = AutoImageProcessor.from_pretrained(self.model_name)
        model = AutoModel.from_pretrained(self.model_name)
        inputs = processor(self.dummy_image, return_tensors="pt")
        outputs = model(**inputs)
        self.assertIsNotNone(outputs)
        
    def test_webgpu(self):
        if not HAS_WEBGPU:
            self.skipTest("WebGPU not available")
            processor = AutoImageProcessor.from_pretrained(self.model_name)
            model = AutoModel.from_pretrained(self.model_name)
            inputs = processor(self.dummy_image, return_tensors="pt")
        # WebGPU simulation mode
            os.environ["WEBGPU_SIMULATION"] = "1",
            outputs = model(**inputs)
            self.assertIsNotNone(outputs)
        # Reset environment
            os.environ.pop("WEBGPU_SIMULATION", None)

File: ipfs_accelerate_js/src/model/vision/test_vit_custom.ts
/**
 * Converted from Python: test_vit_custom.py
 * Conversion date: 2025-03-11 04:08:36
 * This file was automatically converted from Python to TypeScript.
 * Conversion fidelity might not be 100%, please manual review recommended.
 */

// WebGPU related imports
import { HardwareBackend } from "../hardware_abstraction";

import * as $1
import * as $1
import * as $1
import * as $1 as np
import ${$1} from "$1"

# Hardware detection
HAS_CUDA = torch.cuda.is_available()
HAS_WEBGPU = "WEBGPU_AVAILABLE" in os.environ

class TestVit(unittest.TestCase):
  $1($2) {
    this.model_name = "vit"
    this.dummy_image = np.random.rand(3, 224, 224)
  
  }
  $1($2) {
    processor = AutoImageProcessor.from_pretrained(this.model_name)
    model = AutoModel.from_pretrained(this.model_name)
    inputs = processor(this.dummy_image, return_tensors="pt")
    outputs = model(**inputs)
    this.assertIsNotnull(outputs)
    
  }
  $1($2) {
    if ($1) {
      this.skipTest("WebGPU !available")
      processor = AutoImageProcessor.from_pretrained(this.model_name)
      model = AutoModel.from_pretrained(this.model_name)
      inputs = processor(this.dummy_image, return_tensors="pt")
    # WebGPU simulation mode
    }
      os.environ["WEBGPU_SIMULATION"] = "1",
      outputs = model(**inputs)
      this.assertIsNotnull(outputs)
    # Reset environment
      os.environ.pop("WEBGPU_SIMULATION", null)